cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_train_hist_oneapi_impl.i

```diff
@@ -506,7 +549,8 @@ algorithmFPType ClassificationTrainBatchKernelOneAPI<algorithmFPType, hist>::com
     algorithmFPType mean = algorithmFPType(0);
     for (size_t i = 0; i < n; i++)
     {
-        int rowInd        = rowsIndHost.get()[i];
+        int rowInd = rowsIndHost.get()[i];
+        DAAL_ASSERT(rowInd < nRows);
         size_t prediction = mTreeHelper.predict(t, &x[rowInd * nFeatures]);
         oobBufHost.get()[rowInd * _nClasses + prediction]++;
```

Comment:
Is prediction less than nClasses?

Reply:
yes, prediction is an index in [0, _nClasses)